### PR TITLE
pulumiTest Integration tests

### PR DESCRIPTION
**Pull request by VenelinMartinov**
This refactors the cross-tests to expose an integration test module `pulCheck` for use outside of cross-tests.

It allows us to easily write integration tests where we specify a TF schema, which gets bridged and use the pulumiTest framework for the test itself.

### CODE REVIEW

**Comment by t0yv0** [comment]
https://github.com/pulumi/pulumi-terraform-bridge/blob/master/pkg/tests/cross-tests/input_check.go#L132 already using it. Whatever is most convenient.

**Comment by t0yv0** [comment]
Yes, it would. We have a few ProgramTest tests but rebuilding the underlying provider is a bit of a hassle. But it would be really useful to have a quick stencil way to define a quick schema, program in one place and exercise it end to end, under Pulumi and/or TF, with debugging attached by default. Streamlining this would help contributors adding integration tests.

File: pkg/tests/cross-tests/pu_test.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	"github.com/pulumi/providertest/providers"
+	"github.com/pulumi/providertest/pulumitest"
+	"github.com/pulumi/providertest/pulumitest/opttest"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge/tokens"
+	shimv2 "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v2"
+	"github.com/stretchr/testify/require"
+)
+
+func pulTest(t *testing.T, providerName string, resMap map[string]*schema.Resource, program string) *pulumitest.PulumiTest {
```
Comment (iwahbe) [inline pkg/tests/cross-tests/pu_test.go:20]: The file is called `pu_test.go`, this function is `pulTest`. I don't think we need to abbreviate here, but we certainly shouldn't abbreviate inconsistently.

```suggestion
func pulumiTest(t *testing.T, providerName string, resMap map[string]*schema.Resource, program string) *pulumitest.PulumiTest {
```

**Review by iwahbe — APPROVED** [review]
This looks really helpful to me.

File: pkg/tests/cross-tests/defaults.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package crosstests
+
+var (
```
Comment (iwahbe) [inline pkg/tests/cross-tests/defaults.go:3]: I have trouble imagining that these will change.
```suggestion
const (
```

File: pkg/tests/pulcheck/pulcheck.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag/colors"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/rpcutil"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"gotest.tools/assert"
+)
+
+// This is an experimental API.
+func EnsureProviderValid(t T, tfp *schema.Provider) {
```
Comment (iwahbe) [inline pkg/tests/pulcheck/pulcheck.go:33]: Do we want these functions to be called from from outside of `cross-tests`?

If not, should we put them in an `internal` provider.

Comment (VenelinMartinov) [reply pkg/tests/pulcheck/pulcheck.go]: good point, thanks!

**Pull request merged**